src/privileged/vpnRecommender/api.js

```diff
@@ -0,0 +1,562 @@
+const MAX_NOTIFICATION_COUNT = 3;
+
+const VPN_LANDING_PAGE_URL = "https://premium.firefox.com/vpn/";
+const VPN_LANDING_PAGE_DEFAULT_PARAMS = {
+  "utm_source": "firefox-browser",
+  "utm_medium": "firefox-browser",
+  "utm_campaign": "shield_vpn_1",
+};
+
+const log = function(...args) {
+  if (!Preferences.get(DEBUG_MODE_PREF)) return;
+  console.log(...args);
+};
+
+const DOORHANGER_MESSAGES = {
+  "captive-portal": {
+    header: "It appears you are browsing on an unsecured wireless network.",
+    text: "Firefox has teamed up with ProtonVPN to provide you with a private and secure internet connection, no matter where you are.",
+  },
+  "privacy-hostname": {
+    header: "Make Firefox even more secure with ProtonVPN.",
+    text: "Firefox has teamed up with ProtonVPN to provide you with a private and secure internet connection, no matter where you are.",
+  },
+  "streaming-hostname": {
+    header: "Make Firefox even more secure with ProtonVPN.",
+    text: "Firefox has teamed up with ProtonVPN to provide you with a private and secure internet connection, no matter where you are.",
+  },
+  "catch-all": {
+    header: "Make Firefox even more secure with ProtonVPN.",
+    text: "Firefox has teamed up with ProtonVPN to provide you with a private and secure internet connection, no matter where you are.",
+  },
+};
+
+this.vpnRecommender = class extends ExtensionAPI {
+
+  getAPI(context) {
+    const that = this;
+
+    const jsms = {}; // workaround for not being able to use jsms with moz:// urls
+
+    Services.scriptloader.loadSubScript(context.extension.getURL("privileged/vpnRecommender/Doorhanger.jsm"), jsms);
+    Services.scriptloader.loadSubScript(context.extension.getURL("privileged/vpnRecommender/VpnRelatedHostnames.jsm"), jsms);
+    Services.scriptloader.loadSubScript(context.extension.getURL("privileged/vpnRecommender/RecentWindow.jsm"), jsms);
+    Services.scriptloader.loadSubScript(context.extension.getURL("privileged/vpnRecommender/EveryWindow.jsm"), jsms);
+
+    this.Doorhanger = jsms.Doorhanger;
+    this.VpnRelatedHostnames = jsms.VpnRelatedHostnames;
+    this.RecentWindow = jsms.RecentWindow;
+    this.EveryWindow = jsms.EveryWindow;
+
+    this.addCleanUpFunction(() => {
+      Cu.unload(context.extension.getURL("privileged/vpnRecommender/Doorhanger.jsm"));
+      Cu.unload(context.extension.getURL("privileged/vpnRecommender/VpnRelatedHostnames.jsm"));
+      Cu.unload(context.extension.getURL("privileged/vpnRecommender/RecentWindow.jsm"));
+      Cu.unload(context.extension.getURL("privileged/vpnRecommender/EveryWindow.jsm"));
+    });
+
+    this.extensionUrl = context.extension.getURL();
+
+    const EventManager = ExtensionCommon.EventManager;
+
+    this.listenForAddonDisableOrUninstall(context.extension.id); // https://github.com/mozilla/shield-studies-addon-utils/issues/247
+
+    AddonManager.addAddonListener(this);
+    return {
+      experiments: {
+        vpnRecommender: {
+          echo(str) {
```

Comment:
What is this for?

Reply:
i used it for manual testing. not needed. removed.